### PR TITLE
docs: update the doc with some things that were missing

### PR DESCRIPTION
**Pull request by andrew-hirata-playeveryware**
Added some things that were missing from the standards docs:
    * Add C++ specific style standards
    * Add Function prefix documentation
    * Add example of platform specific scopes
    * Add example of of release scheme

### CODE REVIEW

File: docs/standards.md [hunk 1]
```diff
@@ -8,11 +8,11 @@
...
 Each source code file needs to contain the license which it is licensed under.
 
-### For C / C++ / C#:
+### For C / C++ :
 
-### Variables
+#### Variables
+* structs and classes should be PascalCase; initialisms should all be the same case i.e. URL not Url.
 * static variables should start with `s_`, globals with `g_`.
-* Objects of type `struct` should be PascalCase; Abbreviations should always be uppercase (for Example favor `URL` over `url` when as a component of a named type).
```
Comment (paulhazen) [inline docs/standards.md:15]: Seems like this was a more comprehensive and clearer version of what you replaced it with (except it was missing the fact that it should also apply to classes).

File: docs/standards.md [hunk 2]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
...
 Exported function names are prefixed, and lowercase snake case.
+
+The prefix used is based on the name of the dynamic library and the functionality it provides.
+```
+  DynamicLibraryLoaderHelper -> DLLH_
+  Memory Allocation -> Mem_
+  MicrophoneUtility -> MicrophoneUtility_
+```
+
+Prefacing the definition of the function with `FUN_EXPORT` is doen to allow for the code to be defined and exported under
```
Comment (paulhazen) [inline docs/standards.md]: doen should be done.

File: docs/standards.md [hunk 3]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
...
+* Objects of type `struct` or classe should be PascalCase; Abbreviations should always be uppercase (for Example favor `URL` over `url` when as a component of a named type).
+* All other variables should camelCase.
+* use 4 spaces to indent.
+* Constants should done in SCREAMING_SNAKE_CASE.
+
 This follows, more or less, Microsoft's C# style [guide](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions).
 
-### Commit Message Style
+
+## Commit Message Style
```
Comment (WispyMouse) [inline docs/standards.md]: Intentionaly going to ## instead of ###?

File: docs/standards.md [hunk 4]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
...
 e.g.
 `FUN_EXPORT(bool) DLLH_unload_library_at_path(void *ctx, void *library_handle)`
 
-The Epic Online Services SDK has it's one standards and style [guide](https://docs.unrealengine.com/4.26/en-US/ProductionPipelines/DevelopmentSetup/CodingStandard/).
+For reference, the Epic Online Services SDK has it's one standards and style [guide](https://docs.unrealengine.com/4.26/en-US/ProductionPipelines/DevelopmentSetup/CodingStandard/).
 
 ### For C#
+
+#### Variables
+* static variables should start with `s_`, globals with `g_`.
```
Comment (paulhazen) [inline docs/standards.md]: Unless they are constants, in which case they should be `ThisCase` (can't remember what that's called)

File: docs/standards.md [hunk 5]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
...
+
 e.g.
 `FUN_EXPORT(bool) DLLH_unload_library_at_path(void *ctx, void *library_handle)`
 
-The Epic Online Services SDK has it's one standards and style [guide](https://docs.unrealengine.com/4.26/en-US/ProductionPipelines/DevelopmentSetup/CodingStandard/).
+For reference, the Epic Online Services SDK has it's one standards and style [guide](https://docs.unrealengine.com/4.26/en-US/ProductionPipelines/DevelopmentSetup/CodingStandard/).
 
 ### For C#
+
+#### Variables
```
Comment (paulhazen) [inline docs/standards.md:46]: Is this a duplicated section?

File: docs/standards.md [hunk 6]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
...
+* static variables should start with `s_`, globals with `g_`.
+* Objects of type `struct` or classe should be PascalCase; Abbreviations should always be uppercase (for Example favor `URL` over `url` when as a component of a named type).
+* All other variables should camelCase.
+* use 4 spaces to indent.
```
Comment (paulhazen) [inline docs/standards.md]: Should be capitalized.

File: docs/standards.md (hunk 4 above)
Comment (paulhazen) [reply docs/standards.md]: also should use the ticks to make it `static` instead of static

File: docs/standards.md [hunk 7]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
...
+* Constants should done in SCREAMING_SNAKE_CASE.
```
Comment (paulhazen) [inline docs/standards.md]: Interesting - this I think is different in C# vs C++

File: docs/standards.md [hunk 8]
```diff
@@ -22,17 +22,37 @@ Additionally, here are some common variable suffixes that are recommended:
 `_f`    for a pointer to a function
 `_ctx`  for context data i.e. data that is passed into a function
 
-### Functions
+#### Functions
 Exported function names are prefixed, and lowercase snake case.
+
+The prefix used is based on the name of the dynamic library and the functionality it provides.
+```
+  DynamicLibraryLoaderHelper -> DLLH_
```
Comment (paulhazen) [inline docs/standards.md:31]: Could this be misleading for circumstances where it's not actually a DLL? Like if it's a `dylib`?

**Review by WispyMouse — APPROVED** [review]

**Review by paulhazen — APPROVED** [review]
Approving - feel free to resolve any conversations I opened, they are all sort of open-handed issues in my mind.

**Pull request merged**